src/assisted_test_infra/test_infra/helper_classes/base_cluster.py

```diff
@@ -0,0 +1,106 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Optional, Union
+
+from assisted_service_client import models
+from junit_report import JunitTestCase
+
+from assisted_test_infra.test_infra import BaseClusterConfig, BaseInfraEnvConfig, Nodes
+from assisted_test_infra.test_infra.helper_classes.entity import Entity
+from assisted_test_infra.test_infra.helper_classes.infra_env import InfraEnv
+from assisted_test_infra.test_infra.tools import static_network
+from service_client import InventoryClient, log
+
+
+class BaseCluster(Entity, ABC):
+    _config: BaseClusterConfig
+
+    def __init__(
+        self,
+        api_client: InventoryClient,
+        config: BaseClusterConfig,
+        infra_env_config: BaseInfraEnvConfig,
+        nodes: Optional[Nodes] = None,
+    ):
+        self._infra_env_config = infra_env_config
+        self._infra_env: Optional[InfraEnv] = None
+
+        super().__init__(api_client, config, nodes)
+
+        # Update infraEnv configurations
+        self._infra_env_config.cluster_id = config.cluster_id
+        self._infra_env_config.openshift_version = self._config.openshift_version
+        self._infra_env_config.pull_secret = self._config.pull_secret
+
+    @property
+    def id(self) -> str:
+        return self._config.cluster_id
+
+    def get_details(self) -> Union[models.infra_env.InfraEnv, models.cluster.Cluster]:
```

Comment:
how can it return an infra-env?

Reply:
it's the method signature defined in the `Entity` base class. The `Entity` base class is inherited by `*Cluster` classes and `InfraEnv` class.

Reply:
I agree that it's confusing